File: java/var-core/src/main/java/com/oselvar/var/core/Parse.java
package com.oselvar.var.core;

/**
 * Top-level entry point of the pure core: {@code scan} the source into blocks, then {@code
 * structure} those blocks into a {@link Ast.VarDoc}.
 *
 * <p>Port of {@code var-core/src/parse.ts}. That file's {@code plugins} parameter is intentionally
 * out of scope here, mirroring {@link Scanner#scan}, which takes no plugins parameter in this port.
 */
public final class Parse {

    private Parse() {}

    /** Parses {@code source} into a {@link Ast.VarDoc}: {@code scan} then {@code structure}. */
    public static Ast.VarDoc parse(String path, String source) {
        return Structurer.structure(path, source, Scanner.scan(source));
    }
}

File: rust/var-core/src/parse.rs
//! Top-level parse entry point: `scan` then `structure` — port of `parse.ts` /
//! `Parse.java`.

use crate::ast::VarDoc;
use crate::{scanner, structurer};

/// Parses `source` into a [`VarDoc`].
pub fn parse(path: &str, source: &str) -> VarDoc {
    structurer::structure(path, source, scanner::scan(source))
}
